crates/bevy_sprite/src/render/mod.rs

```diff
@@ -336,7 +356,11 @@ pub fn prepare_sprites(
     let mut current_batch_handle: Option<Handle<Image>> = None;
     let mut current_batch_colored = false;
     let mut last_z = 0.0;
-    for extracted_sprite in extracted_sprites.sprites.iter() {
+    for (index, _) in extracted_sprites.indexes.iter().enumerate() {
+        let extracted_sprite = extracted_sprites
+            .sprites
+            .get(index)
```

Comment:
This is wrong? If I am reading this correctly...

You are just getting sequential integers from `enumerate`, which means you are getting the extracted sprites from the *unsorted* array *in-order*, which means the sorting of the index array actually had no effect.

I think you need to store the index integer inside a field in the index struct. Generate the successive integers in `push` when extracting. Then read the field from the sorted index array.

Reply:
removed as it was slower